test/README.md

```diff
@@ -0,0 +1,67 @@
+Candid test data
+================
+
+This directory contains Candid test data, to be used to check Candid
+implementations for compliance.
+
+This test suite covers
+
+ * parsing Candid service descriptions
+ * given a Candid type, decoding binary Candid data, asserting success or failure
+ * given a Candid type, decoding textual Candid data, asserting success or failure
+
+Because this test data necessarily has to be host-language agnostic, it is not
+possible to
+
+ * assert the decoded value, as it would be host-language specific
+ * cover encoding, as it would require specifying a host-language value,
```

Comment:
I still would like to cover encoding somehow. Given a textual Candid value and a type, it's at least possible to say if we can encode the value at this type or not. We can also give one possible encoding and use `candiff` to check for equivalence.

Reply:
Would be nice to cover it somehow…

So by giving a candid textual value, you are actually testing “textual value parsing → conversion to host value → encoding to candid binary”, when you really care about the second arrow. Might be ok… but it’s actually equivalent to “ binary data encoding → conversion to host value → encoding to candid binary” in that way. So essentially, this is saying that round-tripping works.

I still dislike dependenies on some tools. But shoudn't it be the case that equivalence of two binary values is the same as “both values decode to the same host value”…

Maybe we need to define a candid binary normal form (similar to what we do in https://github.com/dfinity-lab/motoko/pull/1493) :-)

But continuing that thought I still end up with “host langauges should check round-tripping”: If our _decoding_ tests provide good coverage, then host-langaue round-tripping tests (host value → candid binary → host value; check for equality) should check encoding quite well.